templates/cli/lib/commands/push.js.twig

```diff
@@ -269,19 +270,71 @@ const pushFunction = async ({ functionId, all, yes } = {}) => {
         return func;
     });
 
-    for (let func of functions) {
-        log(`Pushing function ${func.name} ( ${func['$id']} )`)
+    log('Validating functions');
+    // Validation is done BEFORE pushing so the deployment process can be run in async with progress update
+    for (let i = 0; i < functions.length; i++) {
```

Comment:
Is there reason why not `for (let func of functions) {`?

Reply:
I agree with the `for of` approach